backend/src/main/java/gov/cdc/usds/simplereport/api/pxp/PatientSelfRegistrationController.java

```diff
@@ -64,6 +65,8 @@ public void register(@RequestBody PatientSelfRegistration body, HttpServletReque
             ? body.getPhoneNumbers()
             : List.of(new PhoneNumberInput(null, parsePhoneNumber(body.getTelephone())));
 
+    var backwardsCompatibleEmails = new PatientEmailsHolder(body.getEmail(), body.getEmails());
```

Comment:
I feel that `backwardsCompatibleEmails` may be a bit ambiguous, and perhaps could be named something better ? It feels like it makes sense right now in the context of this PR, but several months down the line I feel that I am totally going to forget about this and be very confused as to what that means. 

I could also be entirely misinterpreting this and it is a great name. My current interpretation is that it helps pass in single email addresses when necessary, vs the list of email addresses. We are moving to an email list and grabbing the first email as a default email, and that the backwardsCompatibleEmails are the first of that list. Perhaps the name should be primaryEmail, given my current understanding? Happy to hop in a chat.

Reply:
Here, "backwards compatible" specifically means backwards compatible _with older versions of the GraphQL schema_.

Say we deploy a feature that includes both backend changes and changes to the GraphQL queries sent by the frontend. In some cases, the resolvers & controllers on the backend may _require_ the new queries being performed by the frontend. The problem is that, while the server will begin executing this new code immediately, the user's browser may have cached the previous version of the application and is still sending the "old" GraphQL queries.

So here, if I used the new `emails` argument directly, we would cause server errors for those users since `emails` would unexpectedly be `null`. We have to do a little extra work to make sure this resolver would play nice with an older version of the UI.

That said! You are absolutely right that the meaning of this name will fade over time. I would just propose adding a ticket immediately to fix the schema drift as soon as we're comfortable that everyone is on the new UI version (along the lines of [this issue](https://github.com/CDCgov/prime-simplereport/issues/1912)).

Reply:
Ahh, thank you for the explanation, that sounds great!